### PR TITLE
Site Editor: Prevent opening the navigation sidebar, navigate home instead

### PR DESCRIPTION
**Pull request by Copons**
#### Changes proposed in this Pull Request

* Add a new `wpcom-site-editor` package to ETK, which will contain all the customizations for the site editor.
* Inject a new `CustomNavigationToggleButton` into the Site Editor that simply navigates back home instead of toggling the navigation sidebar.
* Hide the "Browse all templates" button in the document settings dropdown.

This is a temporary measure for the Dotcom FSE Beta to let the user take advantage of most of the Site Editor features, without the navigation sidebar's complexity hindering their efforts.

#### Testing instructions

This needs to be tested in both local and Calypso, to make sure the redirection lands on the appropriate URL according to the current environment.

* Have two sites (local and Dotcom) ready with the Core Site Editor (start [here](https://horizon.wordpress.com/new?flags=gutenboarding/site-editor) to create a new one on Dotcom).
* `cd apps/editing-toolkit && yarn dev --sync`
* On the local site:
  * Open the Site Editor.
  * Click on the navigation toggle button (upper left corner).
  * Make sure the navigation sidebar doesn't open.
  * Check that you are redirected to the wp-admin dashboard.
* On the Dotcom site:
  * Sandbox the site.
  * Open the Site Editor.
  * Click on the navigation toggle button (upper left corner).
  * Make sure the navigation sidebar doesn't open.
  * Check that you are redirected to the Calypso My Home page for the current site.

Related to #54460

### CODE REVIEW

**Review by Addison-Stavlo — COMMENTED** [review]
These changes are looking good to me so far!  Il have more time to test the two environments tomorrow.

> The downside is that I can't seem to find a (clean) way to update the button's aria-label and its tooltip that shows up on hover.

<s>don't say dom selectors, don't say dom selectors, don't say dom selectors</s> 😆  I hope someone has a better idea that dom selectors.  But as this should be a temporary thing, maybe we can live with that? <s>no, don't let it happen.  don't go down this road again</s>

Also, we should probably start discussing what to do with this guy:

Screen Shot 2021-08-03 at 9 44 18 AM

Do we need to rename that button since it will have the same action as leaving the editor to the calypsoCloseUrl now?  Do we have special handling for that button so it leads to somewhere else that makes sense?  Do we remove the templates menu dropdown altogether?  It is sort of geared around things we are currently hiding to new users, but it can be helpful "clear customizations" button and starts vaguely introducing people to the idea of a template. 🤷‍♀️

**Comment by Copons** [comment]
> don't say dom selectors, don't say dom selectors, don't say dom selectors 😆 I hope someone has a better idea that dom selectors. But as this should be a temporary thing, maybe we can live with that? no, don't let it happen. don't go down this road again

I've tried the DOM way, it's filed under the "I can't seem to find a (clean) way". 🤣 
Basically React re-renders would overwrite the element's `aria-label` attribute changes, unless I keep doing them forever (or, like, for a while) in an interval. 
Also the tooltip that shows up is all-React: that it's rendered in the general popover slot without any clues that relate it to the toggle button. I'd be ok to hide it via CSS, but I don't think there's a good selector for that.

> Also, we should probably start discussing what to do with this guy:
> 
> Screen Shot 2021-08-03 at 9 44 18 AM
> 
> Do we need to rename that button since it will have the same action as leaving the editor to the calypsoCloseUrl now? Do we have special handling for that button so it leads to somewhere else that makes sense? Do we remove the templates menu dropdown altogether? It is sort of geared around things we are currently hiding to new users, but it can be helpful "clear customizations" button and starts vaguely introducing people to the idea of a template. 🤷‍♀️

Oh good point, I totally forgot about that guy.
I think it's ok to just hide the "browse all" button via CSS for now.
"Clear customizations" should stay, instead.

**Comment by Addison-Stavlo** [comment]
> Basically React re-renders would overwrite the element's aria-label attribute changes, unless I keep doing them forever (or, like, for a while) in an interval.

I think we have used mutation oberservers in the past (like for dotcom fse block inserter) for this sort of thing, to re-apply the changes after react rerenders and ruins your customization.

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-site-editor/navigation-toggle/custom-navigation-toggle-button.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+			window.top.location.href = calypsoCloseUrl;
+		} else {
+			window.location.href = './index.php';
+		}
+	};
+
+	return (
+		<Button
+			className="edit-site-navigation-toggle__button has-icon"
+			label={ __( 'Back to Dashboard', 'full-site-editing' ) }
```
Comment (a8ci18n) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-site-editor/navigation-toggle/custom-navigation-toggle-button.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Back to dashboard' )` *ES Score: 8*

**Review by Addison-Stavlo — COMMENTED** [review]
Tested this locally and on dotcom.  The toggle now exits the editor as expected and the all templates button is hidden on both platforms.

Also opened https://github.com/Automattic/wp-calypso/pull/55558 to skip nav sidebar tests on dotcom once this is deployed.

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-site-editor/navigation-toggle/custom-navigation-toggle-button.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { Button } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { Icon, wordpress } from '@wordpress/icons';
+
+export default function CustomNavigationToggleButton() {
+	const siteIconUrl = useSelect( ( select ) => {
+		const { getEntityRecord } = select( 'core' );
+		return getEntityRecord( 'root', '__unstableBase', undefined )?.site_icon_url;
+	} );
```
Comment (Addison-Stavlo) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-site-editor/navigation-toggle/custom-navigation-toggle-button.js]: This all seems to work well, but I wonder if we can avoid having to build and inject the entire new component?

Cant we just select `.edit-site-navigation-toggle` and add a 'click' event listener that does the same as the onClick in this component?  I tried this in the console and it seemed to work as well:

`document.querySelector( '.edit-site-navigation-toggle' ).addEventListener( 'click', ( ev ) => { ev.preventDefault(); ev.stopPropagation(); window.location.href = './index.php' } )`

Comment (Copons) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-site-editor/navigation-toggle/custom-navigation-toggle-button.js]: Done, thanks for the suggestion!
MutationObserver + addEventListener = PROFIT!

**Review by Addison-Stavlo — APPROVED** [review]
Looks great!  Tests well for me on dotcom and local! 🎉

**Comment by Copons** [comment]
Just a note that this is currently on hold (as its twin PR #55558) until the FSE Beta is closer to rollout.

**Comment by vindl** [comment]
Screenshot 2021-09-08 at 19 41 47

Not sure if it's directly related, but the Document Settings dropdown content looks a bit borked after Browse All Templates removal.

I'm wondering if we even need it in this state, given that the same information is present in Template toolbar.

Screenshot 2021-09-08 at 19 45 33

cc @onuro

**Comment by Addison-Stavlo** [comment]
@vindl - good catch, i think we should definitely clean that up.

One thing the dropdown does add is the "clear customizations" button that becomes available after the template is updated and saved from its default form:

Screen Shot 2021-09-09 at 7 39 37 AM

**Pull request merged**